### PR TITLE
Pb-721 show share whiteboard link and add save whiteboard as pdf feature

### PR DESCRIPTION
**Pull request by nurjinjafar**
@skolmer  aaaand... i guess my IDE reformated the same files your IDE formated. i don't see any changes locally to not push them. so here you go.
could you please check if i overwrote any of your changes?!

### CODE REVIEW

**Comment by skolmer** [comment]
> 
> 
> @skolmer aaaand... i guess my IDE reformated the same files your IDE formated. i don't see any changes locally to not push them. so here you go.
> could you please check if i overwrote any of your changes?!

IDE fight 😁 I will check tomorrow

File: src/js/main.js [hunk 1]
```diff
@@ -404,6 +404,34 @@ function initWhiteboard() {
...
+            .click(function () {
+                whiteboard.getPdfData().then(function (canvas) {
+                    var myImage = canvas.toDataURL("image/png");
+                    // Adjust width and height
+                    var imgWidth = (canvas.width * 60) / 240;
+                    var imgHeight = (canvas.height * 70) / 240;
+                    // jspdf changes
+                    var pdf = new jsPDF("p", "mm", "a4");
+                    pdf.addImage(myImage, "png", 15, 2, imgWidth, imgHeight); // 2: 19
+                    var w = window.open("about:blank"); //Firefox will not allow downloads without extra window
```
Comment (skolmer) [inline src/js/main.js:419]: Do you have some kind of adblocker or popup blocker installed? I tried this example in my firefox which hast uBlock extension installed and it is working: https://jsfiddle.net/ourcodeworld/rce6nn3z/2/

**Review by skolmer — CHANGES_REQUESTED** [review]

File: src/js/main.js (hunk 1 above)
Comment (nurjinjafar) [reply src/js/main.js:419]: without opening an new window the download won't work even in chrome (in widget context)

**Comment by skolmer** [comment]
does "show share whiteboard link" in the title mean that the user is able again to share the url of the whiteboard? this was disabled because user should stay in the widget for now. The element team is even working on removing the popout button of the widget container.

**Comment by nurjinjafar** [comment]
> does "show share whiteboard link" in the title mean that the user is able again to share the url of the whiteboard? this was disabled because user should stay in the widget for now. The element team is even working on removing the popout button of the widget container.

no only admin will be able to share the whiteboard link. for the user it is still removed. if oyu think it's wanted to be prevented then i can hide it again

**Comment by skolmer** [comment]
> 
> 
> > does "show share whiteboard link" in the title mean that the user is able again to share the url of the whiteboard? this was disabled because user should stay in the widget for now. The element team is even working on removing the popout button of the widget container.
> 
> no only admin will be able to share the whiteboard link. for the user it is still removed. if you think it's wanted to be prevented then i can hide it again

ok, lets keep it then.

I did a small change to calculate how much the image should be scaled to always fit the available height and width of the pdf

**Review by skolmer — APPROVED** [review]

**Pull request merged**